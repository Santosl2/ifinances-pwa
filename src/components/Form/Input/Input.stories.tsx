import { ComponentMeta, ComponentStory } from "@storybook/react";

import { Input } from "./Input";

export default {
  title: "Components/Input",
  component: Input,
} as ComponentMeta<typeof Input>;

const Template: ComponentStory<typeof Input> = (args) => <Input {...args} />;

export const Default = Template.bind({});

Default.args = {
  name: "email",
  label: "E-mail",
  type: "email",
  error: {
    message: "Insira um e-mail v√°lido.",
  },
};
