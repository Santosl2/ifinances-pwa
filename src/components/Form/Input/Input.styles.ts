import styled from "styled-components";

import { InputProps } from "./Input.types";

export const InputWrapper = styled.div<Pick<InputProps, "error">>`
  display: flex;
  flex-direction: column;
  gap: 1rem;

  span {
    font-size: 0.8rem;
    color: var(--red-200);
  }

  ${(props) =>
    !!props.error &&
    `
    label {
       color:  var(--red-200);
    }
  input {
    border-bottom-color: var(--red-200);
    &:focus {
    border-bottom-color: var(--red-200);

    }
  }
  `}

  label {
    cursor: pointer;
    font-family: "Poppins", sans-serif;
    font-weight: 300;
  }
`;
export const InputStyled = styled.input`
  padding: 0.5rem 1rem;
  background-color: transparent;
  border: none;
  border-bottom: 1px solid var(--text-body);
  transition: border 0.2s;
  color: white;

  &:focus {
    outline: none;
    border-bottom-width: 2px;
    border-bottom-color: var(--shape);
  }
`;
