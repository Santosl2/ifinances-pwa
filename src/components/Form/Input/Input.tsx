/* eslint-disable react/function-component-definition */
/* eslint-disable react/jsx-props-no-spreading */
import { forwardRef, ForwardRefRenderFunction } from "react";

import { InputStyled, InputWrapper } from "./Input.styles";
import { InputProps } from "./Input.types";

const InputBase: ForwardRefRenderFunction<HTMLInputElement, InputProps> = (
  { label, name, error, ...rest },
  ref
) => {
  return (
    <InputWrapper error={error}>
      {!!label && <label htmlFor={name}>{label}</label>}
      <InputStyled
        name={name}
        id={name}
        placeholder={label}
        ref={ref}
        {...rest}
      />
      {!!error && <span>{error.message}</span>}
    </InputWrapper>
  );
};

export const Input = forwardRef(InputBase);
