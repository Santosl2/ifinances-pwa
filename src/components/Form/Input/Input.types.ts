import { InputHTMLAttributes } from "react";
import { FieldError } from "react-hook-form";

export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  name: string;
  label?: string;
  error?: FieldError;
}
