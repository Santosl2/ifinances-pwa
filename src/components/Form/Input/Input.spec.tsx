import { render, screen } from "@testing-library/react";
import "jest-styled-components";

import { Input } from "./Input";

describe("Input Test", () => {
  it("should render Input", () => {
    render(<Input name="test" placeholder="Test" />);

    expect(screen.getByPlaceholderText("Test")).toBeInTheDocument();
  });

  it("should render Input", () => {
    render(<Input name="test" />);

    expect(screen.queryByPlaceholderText("Test")).not.toBeInTheDocument();
  });

  it("must have a border bottom when focusing ", async () => {
    render(<Input name="test" placeholder="Test" />);

    const input = screen.getByPlaceholderText("Test");

    input.focus();

    expect(input).toHaveFocus();
    expect(input.style).toHaveProperty("border");
  });

  it("should match Snapshot", () => {
    const { container } = render(<Input name="test" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
