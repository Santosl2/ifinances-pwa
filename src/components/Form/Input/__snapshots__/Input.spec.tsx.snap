// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Test should match Snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

.c0 label {
  cursor: pointer;
  font-family: "Roboto",sans-serif;
}

.c1 {
  padding: 0.5rem 1rem;
  background-color: transparent;
  border: none;
  border-bottom: 1px solid var(--text-body);
  -webkit-transition: border 0.2s;
  transition: border 0.2s;
  color: white;
}

.c1:focus {
  outline: none;
  border-bottom-color: var(--purple);
}

<div
  class="c0"
>
  <input
    class="c1"
    id="test"
    name="test"
  />
</div>
`;
